src/components/views/rooms/Stickerpicker.js

```diff
@@ -130,8 +130,13 @@ export default class Stickerpicker extends React.Component {
 
     _updateWidget() {
         const stickerpickerWidget = WidgetUtils.getStickerpickerWidgets()[0];
+        if (!stickerpickerWidget) {
+            ActiveWidgetStore.delStickerPickerWidget();
+            this.setState({stickerpickerWidget: null, widgetId: null});
+            return;
+        }
 
-        const currentWidget = this.state.stickerpickerWidget;
+        const currentWidget = ActiveWidgetStore.getStickerPickerWidget();
```

Comment:
Hmm, I am not sure I like adding the `getStickerPickerWidget` etc. methods to `ActiveWidgetStore`. It seems like we're mixing too many different concerns into `ActiveWidgetStore`.

Do you only need `getStickerPickerWidget` so that you can look up `currentWidget.content.url`? If so, what if instead `PersistedElement` exposed some way to get access to the React component stored under the persist key?

```
const currentPicker = PersistedElement.get(PERSISTED_ELEMENT_KEY);
const currentWidget = currentPicker.state.stickerpickerWidget;
...
```

(Writing `PersistedElement.get` looks a bit tricky though, since `PersistedElement` allows for multiple child elemented...)

Another alternative could be to track a singleton on `Stickerpicker`. Similar to `UserActivity.sharedInstance`, you could have `Stickerpicker.currentWidget` as a static method for looking up the current widget. (Seems simpler than the `PersistedElement` idea above.)

Reply:
This was a desperate attempt to not create additional code to track a stickerpicker, with the very weak justification of "it's an active widget to be stored". Will consider the options and see what fits best without code sprawl.